src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+
+            // animate toast
+            if (ToastOptions.CloseAnimation != null)
+            {
+                await ToastOptions.CloseAnimation((ToastView)toastView);
+            }
+
+            // remove toast
+            toastContainer.Children.Remove(toastView);
+        }
+
+        private async Task<Grid?> GetToastContainerSettingUpIfNeededAsync()
+        {
+            // get current page
+            var currentPage = GetCurrentContentPage();
+            if (currentPage == null)
+            {
+                return null;
+            }
+
+            // try get toast container
+            var toastContainer = FindByName(currentPage.Id.ToString());
+            if (toastContainer != null) // found toast container
+            {
+                // check opened toasts, can be only one or none
+                var oldToast = toastContainer.Children.FirstOrDefault(w => w.GetType() == typeof(ToastView));
+                if (oldToast != null) // close old toast
+                {
+                    CancellationSource.Cancel();
+                    toastContainer.Children.Remove(oldToast);
+                }
+            }
+            else // no toast container
+            {
+                // create and register toast container
+                toastContainer = new Grid();
+                RegisterName(currentPage.Id.ToString(), toastContainer);
+
+                // old content
+                var oldContent = currentPage.Content;
+
+                // set new content
+                await MainThread.InvokeOnMainThreadAsync(() => { currentPage.Content = toastContainer; });
+                toastContainer.Children.Add(oldContent);
+            }
+
+            return toastContainer;
+        }
+
+        private static ContentPage? GetCurrentContentPage()
+        {
+            if (Application.Current.MainPage is ContentPage contentPage)
+            {
+                return contentPage;
+            }
+
+            if (Application.Current.MainPage is NavigationPage navigationPage)
+            {
+                if (navigationPage.CurrentPage.Navigation.ModalStack.Any())
+                {
+                    return navigationPage.CurrentPage.Navigation.ModalStack.Last() as ContentPage;
+                }
+
+                return navigationPage.CurrentPage as ContentPage;
+            }
+
+            if (Application.Current.MainPage is TabbedPage tabbedPage)
+            {
+                if (tabbedPage.CurrentPage is NavigationPage tabNavigationPage)
+                {
+                    if (tabNavigationPage.CurrentPage.Navigation.ModalStack.Any())
+                    {
+                        return tabNavigationPage.CurrentPage.Navigation.ModalStack.Last() as ContentPage;
+                    }
+
+                    return tabNavigationPage.CurrentPage as ContentPage;
+                }
+
+                return tabbedPage.CurrentPage as ContentPage;
+            }
+
+            return null;
```

Comment:
Throw proper exception? To let the consumer know the constraints of our API, instead of having nothing happen.

Reply:
Done